### ISSUE TITLE
[Godot 4.0] Shader only works if the mesh is pointing towards Z+

### ISSUE THREAD

**Issue by Cyrus-Harding** [OBKF/Fake-Interior-Shader-for-GodotEngine#3]
Mesh pointing towards positive Z:
![image](https://user-images.githubusercontent.com/10121762/112338662-43daff80-8cbf-11eb-93c0-6150a7b40b45.png)

Mesh pointing towards negative Z:
![image](https://user-images.githubusercontent.com/10121762/112338714-4f2e2b00-8cbf-11eb-83fc-6680bae1e046.png)

**Comment by Cyrus-Harding** [comment]
Changing
```vec3 I = normalize(vertexW - CameraPosition);```
to
```vec3 I = -normalize(vertexW - CameraPosition);```
Makes it work pointing towards -Z.

Pointing it towards X- or X+ also makes it not work.

**Comment by OBKF** [comment]
Hello @kimamb ,

I am aware of the issue, and already working on the fix, I am trying to simplify the new implementation so it can perform well in large scenes.

**Comment by Cyrus-Harding** [comment]
Thank you very much.

I am looking forward for the update!

**Comment by OBKF** [comment]
Thank you for using it.

**Comment by OBKF** [comment]
@kimabjorkede Hi, the new update is UP, can you try it?

**Comment by wp2000x** [comment]
Now its like this: 

https://user-images.githubusercontent.com/70962730/196826390-8692fa8b-62f1-4961-9e1f-61429b630343.mp4

**Comment by kataura** [comment]
Yup my friend and I are facing this very same issue. Fixed the hints for godot 4 but the shader seems broken on certain meshes for some reason. The middle building works perfectly but the ones to the side have the exact issue as the above video. Exporting glb from blender? same issue. It works on a csgmesh > planemesh though.

**Comment by wp2000x** [comment]
> Yup my friend and I are facing this very same issue. Fixed the hints for godot 4 but the shader seems broken on certain meshes for some reason. The middle building works perfectly but the ones to the side have the exact issue as the above video. Exporting glb from blender? same issue. It works on a csgmesh > planemesh though.

as far as i remember, it was becaose of the mesh object origin point.
